app/Http/Controllers/Consumables/ConsumableCheckoutController.php

```diff
@@ -86,26 +94,35 @@ public function store(Request $request, $consumableId)
         // Check if the user exists
         if (is_null($user = User::find($assigned_to))) {
             // Redirect to the consumable management page with error
-            return redirect()->route('consumables.checkout.show', $consumable)->with('error', trans('admin/consumables/message.checkout.user_does_not_exist'))->withInput();
+            return redirect()->route('consumables.checkout.show', $consumable)
+                ->with('error', trans('admin/consumables/message.checkout.user_does_not_exist'))
+                ->withInput();
         }
-
-        // Update the consumable data
-        $consumable->assigned_to = e($request->input('assigned_to'));
-
-        for ($i = 0; $i < $quantity; $i++){
-        $consumable->users()->attach($consumable->id, [
-            'consumable_id' => $consumable->id,
-            'created_by' => $admin_user->id,
-            'assigned_to' => e($request->input('assigned_to')),
-            'note' => $request->input('note'),
-        ]);
+        $now = now();
+
+        try {
+            $data = [
+                'consumable_id' => $consumable->id,
+                'created_by' => $admin_user->id,
+                'assigned_to' => $assigned_to,
+                'note' => $request->input('note') ?: null,
+                'created_at' => $now,
+                'updated_at' => $now,
+            ];
+
+            // Update the consumable data
+            $attachData = array_fill(0,$quantity, $data);
+
+            DB::transaction(function () use ($consumable, $attachData, $user, $request) {
+                $consumable->users()->attach($attachData);
+                event(new CheckoutableCheckedOut($consumable, $user, auth()->user(), $request->input('note')));
+            });
+        }catch(\Exception $e){
+           report ($e);
```

Comment:
This is the only thing that we _might_ not want to include. I'll leave it up to @snipe.